modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -146,6 +153,11 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
       _ <- if (!canJoinCluster) NodeStateDoesNotAllowForJoining(nodeState).raiseError[F, Unit]
       else Applicative[F].unit
 
+      address <- toPeer.id.value.toPublicKey.map(_.toAddress)
```

Comment:
I have the same line in my Rewards code so I suggest you to do this:

```scala
  @derive(decoder, encoder, eqv, show, order)
  @newtype
  case class Id(hex: Hex) {
    def toPublicKey[F[_]: Async: SecurityProvider]: F[PublicKey] = hex.toPublicKey

    def toAddress[F[_]: Async: SecurityProvider]: F[Address] = toPublicKey.map(_.toAddress)
  }
```

and just `toPeer.id.toAddress` as such conversion is used very often.